### PR TITLE
[connector/*] generate status from metadata.yaml

### PR DESCRIPTION
**Pull request by mwear**
**Description:** 
This PR adds the ability to generate status from metadata.yaml for connectors. The format for connectors differs in that stability is per connection rather than per signal as it is for other components. In addition to the code to generate status, this uses the count connector as a demonstration of how this will work in the real world.

The auto-generated code looks something like this (also, take a look at the count connector readme):

| Status        |           |
| ------------- |-----------|
| Distributions | [contrib], [sumo] |

[development]: https://github.com/open-telemetry/opentelemetry-collector#development
[contrib]: https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib
[sumo]: https://github.com/SumoLogic/sumologic-otel-collector
## Supported Pipeline Types
| [Exporter Pipeline Type] | [Receiver Pipeline Type] | [Stability Level] |
| ------------------------ | ------------------------ | ----------------- |
| traces | metrics | [development] |
| metrics | metrics | [development] |
| logs | metrics | [development] |

[Exporter Pipeline Type]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/connector/README.md#exporter-pipeline-type
[Receiver Pipeline Type]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/connector/README.md#receiver-pipeline-type
[Stability Level]: https://github.com/open-telemetry/opentelemetry-collector#stability-levels

**Link to tracking Issue:** 
This will help with the connector part of #21213.

**Testing:** <Describe what testing was performed and which tests were added.>
Manual

### CODE REVIEW

File: connector/countconnector/README.md [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
-| ------------------------ | ------------------------ |
-| traces                   | metrics                  |
-| metrics                  | metrics                  |
-| logs                     | metrics                  |
+[development]: https://github.com/open-telemetry/opentelemetry-collector#development
+[contrib]: https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib
+[sumo]: https://github.com/SumoLogic/sumologic-otel-collector
+[Exporter Pipeline Type]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/connector/README.md#exporter-pipeline-type
+[Receiver Pipeline Type]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/connector/README.md#receiver-pipeline-type
+[Stability Level]: https://github.com/open-telemetry/opentelemetry-collector#stability-levels
```
Comment (codeboten) [inline connector/countconnector/README.md:21]: note you can remove the links at the bottom of the doc

File: cmd/mdatagen/templates/readme.connector.md.tmpl [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<!-- status autogenerated section -->
+{{- if len .Status.Stability }}
+| Status        |           |
+| ------------- |-----------|
+{{- $stabilityLevels := keys .Status.Stability }}
+| Stability                | [{{ stringsJoin $stabilityLevels "], [" }}] |
```
Comment (codeboten) [inline cmd/mdatagen/templates/readme.connector.md.tmpl:6]: It feels weird to have a stability level here, and another in the table below. I would love to see the two combined somehow

**Review by codeboten — COMMENTED** [review]
Thanks for taking this up @mwear! Add some comments, would love @djaglowski's input on this as well

**Review by mx-psi — COMMENTED** [review]
If #22962 is merged before this one, we should document this on the schema file

File: cmd/mdatagen/templates/readme.connector.md.tmpl (hunk 2 above)
Comment (djaglowski) [reply cmd/mdatagen/templates/readme.connector.md.tmpl]: I agree but I'm not sure I think this format ends up duplicating information about which signal combinations are supported, and also introduces an awkward concept of `traces_to_metrics`, etc:

| Status        |           |
| ------------- |-----------|
| Stability     | [development]: traces_to_metrics, metrics_to_metrics, logs_to_metrics   |
| Distributions | contrib, sumo |

## Supported Pipeline Types
| [Exporter Pipeline Type] | [Receiver Pipeline Type] |
| ------------------------ | ------------------------ |
| traces | metrics |
| metrics | metrics |
| logs | metrics |


---

I know it diverges from other classes of components but in my opinion it's much clearer to represent this info like this:

| Status        |           |
| ------------- |-----------|
| Distributions | contrib, sumo |

## Supported Pipeline Types
| [Exporter Pipeline Type] | [Receiver Pipeline Type] | Stability |
| ------------------------ | ------------------------ | ---|
| traces | metrics | development |
| metrics | metrics | development |
| logs | metrics | development |

Comment (mwear) [reply cmd/mdatagen/templates/readme.connector.md.tmpl]: I updated the rendering to match this suggestion.

**Review by djaglowski — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
Looks great @mwear, thanks for addressing my comments.

**Pull request merged**